ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureBlobDataAccessor.java

```diff
@@ -268,23 +290,28 @@ public void downloadBlob(BlobId blobId, OutputStream outputStream) throws BlobSt
    * @return The {@link CloudBlobMetadata} if the blob was found, or null otherwise.
    * @throws BlobStorageException
    */
-  public CloudBlobMetadata getBlobMetadata(BlobId blobId) throws BlobStorageException {
-    BlobProperties blobProperties = null;
+  public CloudBlobMetadata getBlobMetadata(BlobId blobId) throws Exception {
```

Comment:
Are there many callers that expect these methods to throw BlobStorageException?

Reply:
It looks like callers of these methods are catching all exceptions and wrapping it in `CloudStorageException` in `toCloudStorageException()` method. But, modified it back to throw sub-class exceptions like `BlobStorageException`, `IOException`.